File: order-service/src/main/java/com/catius/order/controller/OrderController.java
package com.catius.order.controller;

import com.catius.order.controller.dto.OrderCreateRequest;
import com.catius.order.controller.dto.OrderResponse;
import com.catius.order.domain.Order;
import com.catius.order.service.OrderService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.servlet.support.ServletUriComponentsBuilder;

import java.net.URI;

@RestController
@RequestMapping("/api/v1/orders")
@RequiredArgsConstructor
public class OrderController {

    private final OrderService service;

    @PostMapping
    public ResponseEntity<OrderResponse> create(@Valid @RequestBody OrderCreateRequest request) {
        Order order = service.createOrder(request.productId(), request.quantity());
        URI location = ServletUriComponentsBuilder
                .fromCurrentRequestUri()
                .path("/{id}")
                .buildAndExpand(order.getId())
                .toUri();
        return ResponseEntity.created(location).body(OrderResponse.from(order));
    }

    @GetMapping("/{id}")
    public OrderResponse get(@PathVariable Long id) {
        return OrderResponse.from(service.getOrder(id));
    }
}

File: order-service/src/main/kotlin/com/catius/order/controller/OrderController.kt
package com.catius.order.controller

import com.catius.order.controller.dto.CreateOrderRequest
import com.catius.order.controller.dto.OrderResponse
import com.catius.order.domain.usecase.CreateOrderUseCase
import com.catius.order.domain.usecase.FindOrderUseCase
import jakarta.validation.Valid
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/v1/orders")
class OrderController(
    private val createOrderUseCase: CreateOrderUseCase,
    private val findOrderUseCase: FindOrderUseCase,
) {
    @PostMapping
    fun create(@RequestBody @Valid request: CreateOrderRequest): OrderResponse {
        val result = createOrderUseCase.create(request.toCommand())
        return OrderResponse.from(result)
    }

    @GetMapping("/{orderId}")
    fun find(@PathVariable orderId: Long): OrderResponse {
        val result = findOrderUseCase.findById(orderId)
        return OrderResponse.from(result)
    }
}
